### PR TITLE
Automated cluster state integration test + enhanced healthCheck #406

### LINKED ISSUE #406: Health check and automated stress test
A developer will need a reproducible automated test to compare the effects of new features to existing stability and performance. This should serve as the ultimate source of truth beyond the current e2e test. 

A healthy peer should:
1) Continually increase total block created (not stall or remain the same) across a period of a ~3seconds
2) Continually increase total snapshots formed, however on a longer period ~(3mins)
3) lastSnapshotHash should be the same across all peers
4) There should be no emptyHeight metrics across all peers
5) No checkpoint validation failures across all peers

These metrics are available directly on the DAO and currently being served to the metrics page on port 9000.

Goals:
1) Peer health check. This should be an automated set of comparisons between the peer and rest of the network to determine if it is consistent with the global state and is able to contribute at a facilitator (no failures or timeouts). Details of a healthy peer are given above.
2) Automated stress test, testing health of peers undergoing various Transaction rates and peers restarting

Definition of Done:
1) Peer health mechanism that can be used for tickets #106 and #393 
2) Automated stress test, monitoring peer health.

### PR DESCRIPTION
**Pull request by wkoszycki**
Resolves #406

### CODE REVIEW

File: src/it/scala/org/constellation/ClusterHealthCheckTest.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  implicit val materialize: ActorMaterializer = ActorMaterializer()
+  implicit val executionContext: ExecutionContextExecutor = system.dispatcher
+
+  val (ignoreIPs, auxAPIs) = ComputeTestUtil.getAuxiliaryNodes()
+  val apis: Seq[APIClient] = ComputeTestUtil.createApisFromIpFile(
+    System.getenv().getOrDefault("HOSTS_FILE", "hosts-2.txt"),
+    ignoreIPs
+  )
+
+  // For fixing some old bug, revisit later if necessary
```
Comment (Nikolaj-K) [inline src/it/scala/org/constellation/ClusterHealthCheckTest.scala]: some old bug? 🤔

File: src/main/scala/org/constellation/util/HealthChecker.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      val localMetricsResult = checkLocallMetrics(metrics,a.baseURI)
+      if(localMetricsResult.isLeft) return localMetricsResult
+      hashes ++= Set(metrics.getOrElse(Metrics.lastSnapshotHash, "no_snap"))
+      if (hashes.size > 1) return Left(InconsistentSnapshotHash(a.baseURI, hashes))
+
+    }
+    Right(())
+  }
+
+  def checkLocallMetrics(metrics: Map[String, String], nodeId: String): Either[MetricFailure, Unit] = {
```
Comment (buckysballs) [inline src/main/scala/org/constellation/util/HealthChecker.scala]: Don't mean to be pedantic but checkLocallMetrics => checkLocalMetrics

**Comment by buckysballs** [comment]
lgtm, have we tested on the cluster?

File: src/it/scala/org/constellation/ClusterHealthCheckTest.scala (hunk 1 above)
Comment (wkoszycki) [reply src/it/scala/org/constellation/ClusterHealthCheckTest.scala]: Once I will be able to complete test fully, I will remove it to check if it's still relevant

**Comment by wkoszycki** [comment]
@buckysballs checked against cluster but still we need to wait for #404 to be resolved to check the snapshot test logic works properly that's why it's a Draft PR.

**Comment by wkoszycki** [comment]
how to run has been described in `deploy/README.md`

Verified on stable cluster 
```
[info] ClusterHealthCheckTest:
[info] All nodes
[info] - should have valid metrics
[info] All nodes
[info] - should continually increase total block created (not stall or remain the same) across a period
[info] All nodes
[info] - should continually increase total snapshots count
[info] ScalaTest
[info] Run completed in 9 minutes, 4 seconds.
[info] Total number of tests run: 3
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 3, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 3, Failed 0, Errors 0, Passed 3
[success] Total time: 546 s, completed May 10, 2019 11:24:01 AM
```

File: deploy/README.md [hunk 3]
```diff
@@ -18,3 +18,12 @@ for bringing up machines
 `update-hosts.sh <hosts-file> <node-label>`
 
 `redeploy.sh <hosts-file> <node-label>`
+
+Checking active cluster after initialization (requires hosts file in one of locations:
+
+- environmental variable HOSTS_FILE
+- <project_root>/hosts-2.txt
+- <project_root>/terraform/<any_dir>*hosts
```
Comment (marcinwadon) [inline deploy/README.md]: `<any_dir>/hosts`

File: src/it/scala/org/constellation/ClusterHealthCheckTest.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import com.typesafe.scalalogging.Logger
+import org.constellation.util.{APIClient, HealthChecker, Metrics}
+import org.scalatest.{BeforeAndAfterAll, CancelAfterFailure, FlatSpecLike}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.duration.{FiniteDuration, _}
+
+/**
+  *
```
Comment (marcinwadon) [inline src/it/scala/org/constellation/ClusterHealthCheckTest.scala]: ?

**Pull request merged**